shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/controllers/BridgeExecutorController.java

```diff
@@ -82,27 +98,66 @@ public List<EventSource> prepareEventSources(EventSourceContext<BridgeExecutor>
 
     @Override
     public UpdateControl<BridgeExecutor> reconcile(BridgeExecutor bridgeExecutor, Context context) {
-        LOGGER.info("Create or update BridgeProcessor: '{}' in namespace '{}'", bridgeExecutor.getMetadata().getName(), bridgeExecutor.getMetadata().getNamespace());
+        LOGGER.info("Create or update BridgeProcessor: '{}' in namespace '{}'",
+                bridgeExecutor.getMetadata().getName(),
+                bridgeExecutor.getMetadata().getNamespace());
+
+        // Set a "start" time on the creation
+        BridgeExecutorStatus status = bridgeExecutor.getStatus();
+        if (status.getConditionByType(ConditionTypeConstants.PROGRESSING).isPresent()) {
+            if (status.getConditionByType(ConditionTypeConstants.PROGRESSING).get().getStatus() == ConditionStatus.Unknown) {
+                status.markConditionTrue(ConditionTypeConstants.PROGRESSING, ConditionReasonConstants.DEPLOYMENT_INITIATED);
+                return UpdateControl.updateStatus(bridgeExecutor).rescheduleAfter(executorPollIntervalSeconds);
+            }
+        }
+
+        if (isTimedOut(bridgeExecutor)) {
+            notifyManagerOfFailure(bridgeExecutor,
+                    new ProvisioningTimeOutException(String.format(ProvisioningTimeOutException.TIMEOUT_FAILURE_MESSAGE,
+                            bridgeExecutor.getClass().getSimpleName(),
+                            bridgeExecutor.getSpec().getId())));
+            status.markConditionFalse(ConditionTypeConstants.READY);
+            status.markConditionFalse(ConditionTypeConstants.AUGMENTATION);
+            status.markConditionFalse(ConditionTypeConstants.PROGRESSING);
+            return UpdateControl.updateStatus(bridgeExecutor);
+        }
 
         Secret secret = bridgeExecutorService.fetchBridgeExecutorSecret(bridgeExecutor);
 
         if (secret == null) {
             LOGGER.info("Secrets for the BridgeProcessor '{}' have been not created yet.", bridgeExecutor.getMetadata().getName());
             bridgeExecutor.getStatus().markConditionFalse(ConditionTypeConstants.READY);
             bridgeExecutor.getStatus().markConditionTrue(ConditionTypeConstants.AUGMENTATION, ConditionReasonConstants.SECRETS_NOT_FOUND);
-            return UpdateControl.updateStatus(bridgeExecutor);
+            return UpdateControl.updateStatus(bridgeExecutor).rescheduleAfter(executorPollIntervalSeconds);
+        } else {
+            // Update "Last Transition Time", effectively resetting time-out.
+            status.getConditionByType(ConditionTypeConstants.AUGMENTATION).ifPresent(c -> {
+                if (Objects.equals(c.getReason(), ConditionReasonConstants.SECRETS_NOT_FOUND)) {
+                    status.markConditionTrue(ConditionTypeConstants.PROGRESSING, ConditionReasonConstants.SECRETS_FOUND);
+                }
+            });
```

Comment:
Unfortunately this snippet is not going to be used as the flow would be the following: 

- If `secret == null` -> update status in CRD and reschedule (in this case the changes are written on the CRD when you call `return UpdateControl.updateStatus(bridgeExecutor).rescheduleAfter(executorPollIntervalSeconds);`)
- if `secret != null` -> **in theory** update the status in the CRD and continue (the status is not really updated on the CRD, it is kept in memory)
 - if `!image.equals(bridgeExecutor.getSpec().getImage())` -> Overwrite the status in the CRD and reschedule (the changes are written on the CRD)
 - if `image.equals(bridgeExecutor.getSpec().getImage())` -> **in theory** Overwrite the status in the CRD and continue (the changes are NOT written on the CRD)
 
In short, all the snippets in the `else` branch of the steps have no effect at all. 

I think we should create specific `conditions` for each step, an example from knative is 
```
    - lastTransitionTime: '2022-06-23T13:34:35Z'
      status: 'True'
      type: Addressable
    - lastTransitionTime: '2022-06-23T13:34:32Z'
      reason: Config map knative-eventing/kafka-broker-brokers-triggers updated
      status: 'True'
      type: ConfigMapUpdated
    - lastTransitionTime: '2022-06-23T13:34:32Z'
      status: 'True'
      type: ConfigParsed
    - lastTransitionTime: '2022-08-28T00:36:46Z'
      status: 'True'
      type: DataPlaneAvailable
    - lastTransitionTime: '2022-09-09T04:50:02Z'
      status: 'True'
      type: ProbeSucceeded
    - lastTransitionTime: '2022-09-09T04:50:02Z'
      status: 'True'
      type: Ready
    - lastTransitionTime: '2022-06-23T13:34:32Z'
      reason: Topic ob-dev-brdg-0aabd1ed-20cd-4710-baef-4d93afbf98b3 created
      status: 'True'
      type: TopicReady
```

So instead of using `Augmenting` for everything, I think we should keep the status of each step in a specific condition to solve the problem I described above. Please correct me if I'm wrong :) wdyt?

Reply:
@r00ta I've refactored `BridgeIngressController` and removed use of the generic `AUGMENTING` condition to more specific counterparts. Could you have another look and let me know what you think? Two points you should note: (1) I realise the blocks setting `Condition`'s to `True` are only "in memory" but they are written to the CR when the following step first fails (or the reconciliation is complete). This way I avoid delaying reconciliation (by returning a Status Update early) just to set a `Condition` to `True`, (2) I only update a `Condition` value if it has changed (I could move this to the `CustomResourceStatus` when I make similar changes to `BridgeExecutorController`). This is to avoid updating the `lastTransitionTime` every reconcile loop due to the idempotency of k8s.

Reply:
Looks great :+1: